src/components/menu-item/menu-item.jsx

```diff
@@ -0,0 +1,147 @@
+import React, { useState, createContext, useContext } from 'react';
+import { motion, AnimatePresence } from 'framer-motion';
+import { ChevronDown } from 'lucide-react';
+import { cn } from '@/utilities/functions';
+
+const MenuContext = createContext();
+const useMenuContext = () => useContext( MenuContext );
+
+const Menu = ( { size = 'md', children, className } ) => {
+	const baseClasses = 'w-64 flex flex-col bg-background-primary p-2';
+
+	return (
+		<MenuContext.Provider value={ { size } }>
+			<div className={ cn( baseClasses, className ) }>
+				{ children }
+			</div>
+		</MenuContext.Provider>
+	);
+};
+
+const MenuList = ( {
+	heading,
+	arrow = false,
+	open: initialOpen = false,
+	onClick,
+	children,
+	className,
+} ) => {
+	const [ isOpen, setIsOpen ] = useState( initialOpen );
+	const { size } = useMenuContext();
+
+	const baseClasses = 'text-text-primary bg-transparent cursor-pointer flex justify-between items-center p-1 gap-1';
+
+	const sizeClasses = {
+		sm: 'text-xs',
+		md: 'text-sm',
+	}?.[ size ];
+	const iconSizeClasses = {
+		sm: 'size-4',
+		md: 'size-5',
+	}?.[ size ];
+
+	const handleToggle = () => {
+		setIsOpen( ! isOpen );
+		if ( onClick ) {
+			onClick( ! isOpen );
+		}
+	};
+
+	return (
+		<div className="p-2">
+			<div
+				role="button"
+				tabIndex="0"
+				onClick={ handleToggle }
+				onKeyPress={ ( e ) => {
+					if ( e.key === 'Enter' || e.key === ' ' ) {
+						handleToggle();
+					}
+				} }
```

Comment:
```suggestion
				onKeyPress={ ( event ) => {
					if ( event.key === 'Enter' || event.key === ' ' ) {
						handleToggle();
					}
				} }
```

Use meaningful variable names instead of `i`, and `e`.